ext/dom/php_dom.c

```diff
@@ -1220,11 +1227,12 @@ bool dom_has_feature(zend_string *feature, zend_string *version)
 xmlNode *dom_get_elements_by_tag_name_ns_raw(xmlNodePtr nodep, char *ns, char *local, int *cur, int index) /* {{{ */
 {
 	xmlNodePtr ret = NULL;
+	bool match_any = local[0] == '*' && local[1] == '\0';
 
 	while (nodep != NULL && (*cur <= index || index == -1)) {
 		if (nodep->type == XML_ELEMENT_NODE) {
-			if (xmlStrEqual(nodep->name, (xmlChar *)local) || xmlStrEqual((xmlChar *)"*", (xmlChar *)local)) {
-				if (ns == NULL || (!strcmp(ns, "") && nodep->ns == NULL) || (nodep->ns != NULL && (xmlStrEqual(nodep->ns->href, (xmlChar *)ns) || xmlStrEqual((xmlChar *)"*", (xmlChar *)ns)))) {
+			if (match_any || xmlStrEqual(nodep->name, (xmlChar *)local)) {
+				if (ns == NULL || (ns[0] == '\0' && nodep->ns == NULL) || (nodep->ns != NULL && (xmlStrEqual(nodep->ns->href, (xmlChar *)ns) || xmlStrEqual((xmlChar *)"*", (xmlChar *)ns)))) {
```

Comment:
Can't you also mimic the ``match_any`` optimization here for ``xmlStrEqual((xmlChar *)"*", (xmlChar *)ns)`` ?

Reply:
By doing this optimisation I actually found a bug in the current stable branches. I'll create a PR for that (and tag you).

Reply:
Submitted a PR for the spec compliance issue: GH-11343